### ISSUE TITLE
VS Error List doesn't show errors from build failure

### ISSUE THREAD

**Issue by ardalis** [dotnet/roslyn-project-system#1185]
Build output lists a couple of errors:
![image](https://cloud.githubusercontent.com/assets/782127/21996565/dccb7caa-dbf8-11e6-8133-f62fc98fe85b.png)

Error List doesn't show these errors (just other unrelated warnings):
![image](https://cloud.githubusercontent.com/assets/782127/21996611/1d569d72-dbf9-11e6-8466-6b1eb5cf931c.png)

Restarting Visual Studio to see if it's helpful... fixes the problem.
![image](https://cloud.githubusercontent.com/assets/782127/21996679/6c9b64b2-dbf9-11e6-9ac9-febb0a4dd58a.png)

**Comment by mavasani** [comment]
@ardalis few quick questions:

1. Do you have Full solution analysis enabled? https://msdn.microsoft.com/en-us/library/mt709421.aspx
2. If not, then we will only display intellisense diagnostics for open files. Do you recall if you had ValueObject.cs open when you didn't see the errors in the error list?
3. Can you try change the error list combo box from "Build + Intellisense" to "Build" and verify you can see the errors?

**Comment by ardalis** [comment]
1. No it isn't. I just checked. This was a new install of VS2017.
2. Yes, the file was open. I didn't show it in my first 2 screenshots but the same files were open before and after the VS restart.
3. I've since fixed the errors. Causing them to reappear (by commenting out this line in my project file: 
    <PackageReference Include="System.Reflection.TypeExtensions" Version="4.3.0" />
)
seems to work as expected. I'm not immediately able to repro the behavior where the errors aren't shown in the Error List window.
